docs/advanced_examples/KNearestNeighbors.ipynb

```diff
@@ -0,0 +1,431 @@
+       "      <td id=\"T_3c7a2_row5_col1\" class=\"data row5 col1\" >[0, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row5_col2\" class=\"data row5 col2\" >0</td>\n",
+       "      <td id=\"T_3c7a2_row5_col3\" class=\"data row5 col3\" >[0, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row5_col4\" class=\"data row5 col4\" >0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th id=\"T_3c7a2_level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
+       "      <td id=\"T_3c7a2_row6_col0\" class=\"data row6 col0\" >5.437388</td>\n",
+       "      <td id=\"T_3c7a2_row6_col1\" class=\"data row6 col1\" >[1, 1, 1]</td>\n",
+       "      <td id=\"T_3c7a2_row6_col2\" class=\"data row6 col2\" >1</td>\n",
+       "      <td id=\"T_3c7a2_row6_col3\" class=\"data row6 col3\" >[1, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row6_col4\" class=\"data row6 col4\" >1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th id=\"T_3c7a2_level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
+       "      <td id=\"T_3c7a2_row7_col0\" class=\"data row7 col0\" >4.737523</td>\n",
+       "      <td id=\"T_3c7a2_row7_col1\" class=\"data row7 col1\" >[1, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row7_col2\" class=\"data row7 col2\" >1</td>\n",
+       "      <td id=\"T_3c7a2_row7_col3\" class=\"data row7 col3\" >[0, 1, 1]</td>\n",
+       "      <td id=\"T_3c7a2_row7_col4\" class=\"data row7 col4\" >1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th id=\"T_3c7a2_level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
+       "      <td id=\"T_3c7a2_row8_col0\" class=\"data row8 col0\" >5.163767</td>\n",
+       "      <td id=\"T_3c7a2_row8_col1\" class=\"data row8 col1\" >[1, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row8_col2\" class=\"data row8 col2\" >1</td>\n",
+       "      <td id=\"T_3c7a2_row8_col3\" class=\"data row8 col3\" >[1, 1, 1]</td>\n",
+       "      <td id=\"T_3c7a2_row8_col4\" class=\"data row8 col4\" >1</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th id=\"T_3c7a2_level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
+       "      <td id=\"T_3c7a2_row9_col0\" class=\"data row9 col0\" >4.256865</td>\n",
+       "      <td id=\"T_3c7a2_row9_col1\" class=\"data row9 col1\" >[1, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row9_col2\" class=\"data row9 col2\" >1</td>\n",
+       "      <td id=\"T_3c7a2_row9_col3\" class=\"data row9 col3\" >[1, 1, 0]</td>\n",
+       "      <td id=\"T_3c7a2_row9_col4\" class=\"data row9 col4\" >1</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n"
+      ],
+      "text/plain": [
+       "<pandas.io.formats.style.Styler at 0x7f61c466f4f0>"
+      ]
+     },
+     "execution_count": 29,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df = pd.DataFrame(\n",
+    "    {\n",
+    "        \"Distance\": distance[:, 0],\n",
```

Comment:
does it also makes sense to have the CML distance ?

Reply:
No.

In the _inference method, we return neither  the distance nor the argsort. But the topk labels.
Otherwise, we will end up with information leakage issues

Reply:
In the _inference method, we return the top-k labels, not the distance or argsort (to prevent information leakage issues).